Source/WTF/wtf/NativePromise.h

```diff
@@ -513,11 +519,9 @@ class NativePromise final : public NativePromiseBase, public ConvertibleToNative
 
             m_results[index].emplace(maybeMove(result));
             if (!--m_outstandingPromises) {
-                Vector<Result> results;
-                results.reserveInitialCapacity(m_results.size());
-                for (auto&& result : m_results)
-                    results.append(WTFMove(result.value()));
-
+                Vector<Result> results(m_results.size(), [&](size_t index) {
+                    return WTFMove(*m_results[index]);
+                });
                 m_producer->resolve(WTFMove(results));
                 m_producer = nullptr;
                 m_results.clear();
```

Comment:
I would be tempted to move m_results before the m_producer->resolve call for extra safety.
It will make sure m_results never gets used in the state where it contains moved items.